copy.lua

```diff
@@ -0,0 +1,417 @@
+                        end
+                        local symbol=digSymbols[tile.dig]
+                        if self.option~='normal' then
+                            symbol=' '
+                        end
+                        dc:pen(fg,bg)
+                        paintMapTile(dc, vp, cursor2, xyz2pos(df.global.cursor.x+x+offsetX,df.global.cursor.y+y+offsetY,df.global.cursor.z), symbol, fg)
+                    end
+                
+                    
+                    --
+                end
+            end
+        end
+        
+    end
+end
+
+function CopyUI:onRenderBody(dc)
+    self:renderOverlay(cursor)
+    
+
+    dc:clear():seek(1,1):pen(COLOR_WHITE):string("Copy UI - "..self.state:gsub("^%a",function(x)return x:upper()end))
+    dc:seek(2,3)
+    local cursor = guidm.getCursorPos()
+    
+    
+
+    
+    if self.state=='paste' then
+        dc:key('CUSTOM_S'):string(": Set Clipboard",COLOR_GREY)
+        
+--
+        dc:newline():newline(1)
+        dc:key('CUSTOM_H'):string(": Flip Horizontal",COLOR_GREY) dc:newline(1)
+        dc:key('CUSTOM_V'):string(": Flip Vertical",COLOR_GREY) dc:newline(1)
+        dc:key('CUSTOM_R'):string(": Rotate 90",COLOR_GREY) dc:newline(1)
+        dc:key('CUSTOM_T'):string(": Rotate -90",COLOR_GREY) dc:newline(1)
+        dc:key('CUSTOM_G'):string(": Cycle Corner",COLOR_GREY) dc:newline(1)
+        dc:key('CUSTOM_I'):string(": Invert",COLOR_GREY) dc:newline(1)
+        dc:key('CUSTOM_C'):string(": Convert to...",COLOR_GREY) dc:newline(1)
+        dc:newline(1)
+        dc:key('CUSTOM_E'):string(": "..(self.option=='erase' and "Erasing" or "Erase"),self.option=='erase' and COLOR_RED or COLOR_GREY) dc:newline(1) --make red
+        dc:key('CUSTOM_X'):string(": "..(self.option=='construction' and "Canceling Contructions" or "Cancel Constructions"),self.option=='construction' and COLOR_GREEN or COLOR_GREY) dc:newline(1) --make red
+        dc:newline():newline(1)
+        dc:key('CUSTOM_B'):string(": Blink Clipboard",self.blink and COLOR_WHITE or COLOR_GREY) dc:newline(1)
+        dc:newline()
+    elseif self.state=='mark' then
+        if self.buffer==nil then
+            dc:string("Select two corners.")
+        end
+        dc:newline():newline(1)
+        dc:key('CUSTOM_P'):string(": Cull Selections",self.cull and COLOR_WHITE or COLOR_GREY)
+    elseif self.state=='convert' then
+        dc:key('CUSTOM_D'):string(": Mine",COLOR_GREY) dc:newline(2)
+        dc:key('CUSTOM_H'):string(": Channel",COLOR_GREY) dc:newline(2)
+        dc:key('CUSTOM_U'):string(": Up Stair",COLOR_GREY) dc:newline(2)
+        dc:key('CUSTOM_J'):string(": Up Stair",COLOR_GREY) dc:newline(2)
+        dc:key('CUSTOM_I'):string(": U/D Stair",COLOR_GREY) dc:newline(2)
+        dc:key('CUSTOM_R'):string(": Up Ramp",COLOR_GREY) dc:newline(2)
+         dc:newline(1)
+        dc:string("To undesignate use the erase command",COLOR_WHITE)
+    end
+    dc:newline(2)
+    dc:newline(1)
```

Comment:
You don't really need the 2 there - dc:newline():newline(1) would be equivalent.
Maybe newline() needs a second `dy` parameter to skip multiple lines, since that seems to be used fairly often...